### PR TITLE
fix: ImageMagick handling

### LINKED ISSUE #71: Installed a fresh lsix on macOS with homebrew, got a IMv7 warning
When running lsix I get this warning very often:
```
WARNING: The convert command is deprecated in IMv7, use "magick"
```

version of Imagemagick currently (2024-06-06) in Homebrew:
```
imagemagick: stable 7.1.1-33
```

macOS 14.5

### PR DESCRIPTION
**Pull request by sammcj**
- Fixes #71
- Improves comparison logic
- Linting

### CODE REVIEW

**Comment by thomasmerz** [comment]
I tested it and this fixes #71 👍🏼

**Comment by esposm03** [comment]
This doesn't work with ImageMagick v6

imagemagickv6

File: lsix [hunk 1]
```diff
@@ -133,7 +133,7 @@ autodetect() {
 	# Attempt to set the number of colors to 256.
 	# This will work for xterm, but fail on a real vt340.
 	IFS=";"  read -a REPLY -s -t ${timeout} -d "S" -p $'\e[?1;3;256S' >&2
-	[[ ${REPLY[1]} == "0" ]] && numcolors=${REPLY[2]}
+	[[ ${REPLY[1]} == "0" && ${REPLY[2]} -gt 0 ]] && numcolors=${REPLY[2]}
```
Comment (hackerb9) [inline lsix]: What terminal requires this change?

File: lsix [hunk 2]
```diff
@@ -230,20 +230,24 @@ main() {
...
-	    onerow[len++]="-label"
-	    onerow[len++]=$(processlabel "$1")
-	    onerow[len++]="file://$1"
-	    shift
-	done
-	montage "${onerow[@]}"  $imoptions gif:-  \
-	    | convert - -colors $numcolors sixel:-
-    done
+    # Create and display montages one row at a time.
+		while [ $# -gt 0 ]; do
```
Comment (hackerb9) [inline lsix]: What was the purpose of changing the indentation?

File: lsix [hunk 3]
```diff
@@ -230,20 +230,24 @@ main() {
...
+						len=${#onerow[@]}
+						onerow[len++]="-label"
+						onerow[len++]="$(processlabel "$1")"
+						onerow[len++]="file://$1"
+						shift
+				done
+
+				# Note: The 'montage' command is not affected by this change, so it remains the same.
+				montage "${onerow[@]}" $imoptions gif:- \
+						| magick - -colors $numcolors sixel:-  # Convert to magick here.
```
Comment (hackerb9) [inline lsix]: This would fail on a system with ImageMagick 6 which does not have a Magick command.

File: lsix [hunk 4]
```diff
@@ -29,16 +28,16 @@ tileheight=$tilesize
 #fontfamily=Mincho			# Wide coverage, comes with MS Windows
 
 # Default font size is based on width of each tile in montage.
-fontsize=$((tilewidth/10))
+fontsize=$((tilewidth / 10))
 #fontsize=16		     # (or set the point size directly, if you prefer)
 
-timeout=0.25		    # How long to wait for terminal to respond
-			    # to a control sequence (in seconds).
+timeout=0.25 # How long to wait for terminal to respond
```
Comment (hackerb9) [inline lsix:34]: I do not understand why you would want to reindent comments in this way.

File: lsix [hunk 5]
```diff
@@ -108,144 +107,149 @@ autodetect() {
...
+        exit 1
     fi
 
     # SIXEL SCROLLING (~DECSDM) is now presumed to be enabled.
     # See https://github.com/hackerb9/lsix/issues/41 for details.
 
     # TERMINAL COLOR AUTODETECTION.
     # Find out how many color registers the terminal has
-    IFS=";"  read -a REPLY -s -t ${timeout} -d "S" -p $'\e[?1;1;0S' >&2
+    IFS=";" read -a REPLY -s -t ${timeout} -d "S" -p $'\e[?1;1;0S' >&2
```
Comment (hackerb9) [inline lsix:124]: Can you please explain to me what you were thinking when you made this change?  Multiple spaces helps separate the environment from the command. Why change it?

**Comment by hackerb9** [comment]
I appreciate the work you put in to make `lsix` better, but there are too many changes without explanation.

**Pull request closed without merging**